controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java

```diff
@@ -50,33 +49,24 @@ private AuthHandler getHandler(String handlerName) throws AuthenticationExceptio
     }
 
     /**
-     * API to authenticate and authroize access to a given resource.
```

Comment:
Another option is to not take the `authhandler` method from the creds at all. We can take it from controller config. I think this will give more flexibility. We can also keep the option of choosing the handler by specifying it through `method`. What do you think?

Reply:
You have understood my comment correctly. The current approach in master allows multiple similar plugins to co-exist. With the approach in this PR, one plugin will take up one type of authentication. (e.g. we can not have two plugins supporting `Basic` at the same time). One possible way to fix this is to select the handler through config rather than from the header.

Reply:
After offline discussions, we closed on this. The decision is to not have multiple such plugins co-existing on the same `scheme`. This comment is taken care of.